File: schemaorg-rs/src/ownership_info.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type OwnershipInfoAdditionalTypeFieldEnum = String;
///https://schema.org/OwnershipInfo
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct OwnershipInfo {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/typeOfGood
    #[serde(rename = "typeOfGood")]
    #[serde_as(as = "OneOrMany<_>")]
    pub type_of_good: Vec<OwnershipInfoTypeOfGoodFieldEnum>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<OwnershipInfoMainEntityOfPageFieldEnum>,
    ///https://schema.org/ownedThrough
    #[serde(rename = "ownedThrough")]
    #[serde_as(as = "OneOrMany<_>")]
    pub owned_through: Vec<String>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/acquiredFrom
    #[serde(rename = "acquiredFrom")]
    #[serde_as(as = "OneOrMany<_>")]
    pub acquired_from: Vec<OwnershipInfoAcquiredFromFieldEnum>,
    ///https://schema.org/ownedFrom
    #[serde(rename = "ownedFrom")]
    #[serde_as(as = "OneOrMany<_>")]
    pub owned_from: Vec<String>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<OwnershipInfoAdditionalTypeFieldEnum>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<OwnershipInfoDescriptionFieldEnum>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<OwnershipInfoIdentifierFieldEnum>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<OwnershipInfoSubjectOfFieldEnum>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<OwnershipInfoImageFieldEnum>,
}

File: schemaorg-ts/src/ownership_info.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type OwnershipInfoAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/OwnershipInfo>
 **/
export interface OwnershipInfo {
  "@context": string;
  "@type": "OwnershipInfo";

  /**
   * <https://schema.org/ownedFrom>
   **/
  ownedFrom: all.OneOrMany<string>;

  /**
   * <https://schema.org/acquiredFrom>
   **/
  acquiredFrom: all.OneOrMany<all.OwnershipInfoAcquiredFromFieldEnum>;

  /**
   * <https://schema.org/typeOfGood>
   **/
  typeOfGood: all.OneOrMany<all.OwnershipInfoTypeOfGoodFieldEnum>;

  /**
   * <https://schema.org/ownedThrough>
   **/
  ownedThrough: all.OneOrMany<string>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.OwnershipInfoMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.OwnershipInfoIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.OwnershipInfoDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.OwnershipInfoSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.OwnershipInfoAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.OwnershipInfoImageFieldEnum>;
}
